mhkit/utils.py

```diff
@@ -62,12 +67,27 @@ def get_statistics(data,freq,period=600):
             continue
         else:
             # Get stats
-            time.append(datachunk.index.values[0])
-            means.append(datachunk.mean())
-            maxs.append(datachunk.max())
-            mins.append(datachunk.min())
-            stdev.append(datachunk.std())
-
+            time.append(datachunk.index.values[0]) # time vector
+            maxs.append(datachunk.max()) # maxes
+            mins.append(datachunk.min()) # mins
+            means.append(datachunk.mean()) # means
+            stdev.append(datachunk.std()) # standard deviation
+            # calculate vector averages and std
+            for v in vector_channels:
+                Ux = sum(np.sin(datachunk[v]*np.pi/180))/len(datachunk)
+                Uy = sum(np.cos(datachunk[v]*np.pi/180))/len(datachunk)
+                vector_avg = (90 - np.arctan2(Uy,Ux)*180/np.pi) # number doesnt seem right
+                if vector_avg<0: vector_avg = vector_avg+360
+                elif vector_avg>360: vector_avg = vector_avg-360              
+                means[i][v] = vector_avg # overwrite scalar average for channel
+                magsum = round((Ux**2 + Uy**2)*1e8)/1e8 # round to 8th decimal place
```

Comment:
This is to prevent very rare cases where roundoff error in some data acquisition systems can cause the following calculation to hiccup. Could be unnecessary.